### PR TITLE
Replace env_logger with sp_tracing

### PR DESCRIPTION
**Pull request by Safrout1**
This PR replaces env_logger with sp_tracing because of an issue with env_logger and gum #4660

### CODE REVIEW

File: bridges/snowbridge/pallets/outbound-queue/merkle-tree/Cargo.toml
Comment (AndreiEres) [inline bridges/snowbridge/pallets/outbound-queue/merkle-tree/Cargo.toml:30]: I think env_logger should be removed

**Review by AndreiEres — COMMENTED** [review]
That's ok

File: bridges/snowbridge/pallets/outbound-queue/merkle-tree/Cargo.toml
Comment (Safrout1) [reply bridges/snowbridge/pallets/outbound-queue/merkle-tree/Cargo.toml:29]: removed

**Review by bkchr — COMMENTED** [review]
If `env_logger` is not used anymore anywhere, you should also remove it from the workspace `Cargo.toml` file.

**Comment by Safrout1** [comment]
@bkchr No, it is used in 4 non-test files: 
1 | bridges/relays/utils/src/initialize.rs
2 | polkadot/node/subsystem-bench/src/cli/subsystem-bench.rs
3 | substrate/client/executor/benches/bench.rs
4 | substrate/utils/frame/omni-bencher/src/main.rs

File: substrate/frame/sassafras/src/tests.rs
Comment (acatangiu) [inline substrate/frame/sassafras/src/tests.rs:791]:
```suggestion
```

**Review by acatangiu — COMMENTED** [review]
Can you extend this outside of testing too? Replace env_logger completely?

File: polkadot/xcm/xcm-runtime-apis/tests/fee_estimation.rs
Comment (AndreiEres) [inline polkadot/xcm/xcm-runtime-apis/tests/fee_estimation.rs:44]: From clippy

> omit the `let` binding: `sp_tracing::init_for_tests();`

(Not only here)

**Review by AndreiEres — COMMENTED** [review]
A few comments

**Comment by AndreiEres** [comment]
> 1 | bridges/relays/utils/src/initialize.rs

It's inside `initialize_logger` which used only in `initialize_relay`. I don't see any usages of `initialize_relay` in the code base. @acatangiu wdyt is it ok to replace it with `try_init_simple`?

> 2 | polkadot/node/subsystem-bench/src/cli/subsystem-bench.rs

Here is ok to replace with `try_init_simple`

> 3 | substrate/client/executor/benches/bench.rs
> 4 | substrate/utils/frame/omni-bencher/src/main.rs

We can try to use `try_init_simple` here as well. Let's change and ask guys that put it there if it's ok.

**Comment by acatangiu** [comment]
cc @serban300 ^

**Comment by AndreiEres** [comment]
> I have updated the PR with all test files except for one: substrate/frame/contracts/src/tests.rs
> This one's tests fail when using sp-tracing, I think it has to do with setting the right env, I am still figuring it out. Let me know if the test is the wrong one in that case.

Try `try_init_simple`.

**Comment by serban300** [comment]
> > 1 | bridges/relays/utils/src/initialize.rs
> 
> It's inside `initialize_logger` which used only in `initialize_relay`. I don't see any usages of `initialize_relay` in the code base. @acatangiu wdyt is it ok to replace it with `try_init_simple`?

It is used in parity-bridges-common: https://github.com/paritytech/parity-bridges-common/blob/63ac22d90462d14a1f1e9c36427473fa2fa57cb1/substrate-relay/src/cli/mod.rs#L93

I'm not sure what `try_init_simple()` does, but I think it would be good to keep the current characteristics, for example:
```
	builder.filter_level(log::LevelFilter::Warn);
	builder.filter_module("bridge", log::LevelFilter::Info);
```
The time format is probably not super important.

**Comment by acatangiu** [comment]
> The CI pipeline was cancelled due to failure one of the required jobs. **Job name**: cargo-clippy **Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/6817241

@Safrout1 please fix these remaining issues

File: polkadot/node/network/bitfield-distribution/Cargo.toml
Comment (Safrout1) [reply polkadot/node/network/bitfield-distribution/Cargo.toml:32]: The compiler complained that it was not used.

File: polkadot/node/network/collator-protocol/Cargo.toml
Comment (Safrout1) [reply polkadot/node/network/collator-protocol/Cargo.toml:32]: Same, the compiler complained that it was not used.

File: polkadot/xcm/xcm-runtime-apis/tests/fee_estimation.rs
Comment (Safrout1) [reply polkadot/xcm/xcm-runtime-apis/tests/fee_estimation.rs]: Done

**Comment by Safrout1** [comment]
Addressed all comments to this point

**Comment by Safrout1** [comment]
@acatangiu I have added a fix for the failing job, I am not sure how to re-run it. Or do I need you to re run it manually every time?

File: bridges/relays/utils/src/initialize.rs
Comment (AndreiEres) [inline bridges/relays/utils/src/initialize.rs:21]: I don't like what's happening here :-)
Please revert these changes.

Let's keep only the changes that can be done using the one-liners init_for_tests and try_init_simple in this PR. All other changes should be done in a separate PR.

**Review by AndreiEres — CHANGES_REQUESTED** [review]

File: bridges/relays/utils/src/initialize.rs
Comment (acatangiu) [reply bridges/relays/utils/src/initialize.rs:21]: This is fine IMO, it's clear/simple/easy change aligned with the plan to replace `env_logger` with `sp_tracing`.

@serban300 can make final decision as the code-owner of this file.

**Comment by acatangiu** [comment]
> @acatangiu I have added a fix for the failing job, I am not sure how to re-run it. Or do I need you to re run it manually every time?

Has to be manually approved by a maintainer after each new external commit. I will trigger this for you every time I see new pushes, don't worry.

**Review by acatangiu — APPROVED** [review]

File: bridges/relays/utils/src/initialize.rs
Comment (serban300) [reply bridges/relays/utils/src/initialize.rs:21]: Looks good to me as well

Comment (AndreiEres) [reply bridges/relays/utils/src/initialize.rs:21]: Ok then :-)

**Comment by acatangiu** [comment]
@Safrout1 why the huge `Cargo.lock` update?

**Comment by Safrout1** [comment]
> @Safrout1 why the huge `Cargo.lock` update?

This job here: https://github.com/paritytech/polkadot-sdk/actions/runs/10145114392/job/28051448788
is having a problem with `pub use tracing_subscriber;` however, it is imported in the dependencies IIUC.

I don't know why this is happening, but I am sure I am ignorant enough in rust to do something stupid

**Comment by AndreiEres** [comment]
> > @Safrout1 why the huge `Cargo.lock` update?
> 
> This job here: https://github.com/paritytech/polkadot-sdk/actions/runs/10145114392/job/28051448788 is having a problem with `pub use tracing_subscriber;` however, it is imported in the dependencies IIUC.
> 
> I don't know why this is happening, but I am sure I am ignorant enough in rust to do something stupid

It should be reverted anyway. We can't allow such drastic changes in Cargo.lock.

**Comment by AndreiEres** [comment]
The test fails because it runs with `--no-default-features` but `tracing-subscriber` is optional and under default `std` feature.

**Comment by Safrout1** [comment]
@acatangiu @AndreiEres reverted

**Comment by AndreiEres** [comment]
> @acatangiu @AndreiEres reverted

Thank you, but you still have to handle this broken test (without changing the test).

File: substrate/primitives/tracing/src/lib.rs
Comment (AndreiEres) [inline substrate/primitives/tracing/src/lib.rs:49]:
```suggestion
#[cfg(feature = "std")]
pub use tracing_subscriber;
```

File: bridges/relays/utils/Cargo.toml
Comment (AndreiEres) [inline bridges/relays/utils/Cargo.toml:20]:
```suggestion
sp-tracing = { workspace = true, default-features = true }
```

**Review by AndreiEres — CHANGES_REQUESTED** [review]
Try this one to fix the tests :-)

**Comment by acatangiu** [comment]
@Safrout1 clippy failed again - you can locally run `SKIP_WASM_BUILD=1 cargo clippy --all-targets --locked --workspace --quiet` to check everything locally before pushing.

Or you could do quick check of only crates you are changing: `SKIP_WASM_BUILD=1 cargo clippy --locked -p <crate_name>`

**Comment by AndreiEres** [comment]
@Safrout1 please fix failed checks

**Comment by Safrout1** [comment]
> This PR replaces env_logger with sp_tracing because of an issue with env_logger and gum #4660

@AndreiEres what type of audience and bump do you think I should use here in the prdoc?

**Comment by Safrout1** [comment]
> > This PR replaces env_logger with sp_tracing because of an issue with env_logger and gum #4660
> 
> @AndreiEres what type of audience and bump do you think I should use here in the prdoc?

I think for all tests it should be none, but for the 4 non-test ones should be patch, right?

**Comment by AndreiEres** [comment]
> > This PR replaces env_logger with sp_tracing because of an issue with env_logger and gum #4660
> 
> @AndreiEres what type of audience and bump do you think I should use here in the prdoc?

Node dev

**Comment by acatangiu** [comment]
For any `publish = false` crates, bump can be omitted or set as `none`.

For any publishable crates bump is **mandatory** since code and deps are changed. `patch` bump is good since there are no API or behavioral changes.

**Comment by AndreiEres** [comment]
> what type of audience
Node dev, I suppose

**Comment by Safrout1** [comment]
Hello,

I think the only required check that fails is `continuous-integration/gitlab-check-toml-format`

but whenever I run `taplo format substrate/primitives/tracing/Cargo.toml` on my machine, nothing new happens to the file.

Should I run a different command, the log in the job is also not helpful, WDYT?

@acatangiu @AndreiEres

**Comment by serban300** [comment]
> Hello,
> 
> I think the only required check that fails is `continuous-integration/gitlab-check-toml-format`
> 
> but whenever I run `taplo format substrate/primitives/tracing/Cargo.toml` on my machine, nothing new happens to the file.
> 
> Should I run a different command, the log in the job is also not helpful, WDYT?
> 
> @acatangiu @AndreiEres

Can you also try `taplo format --config .config/taplo.toml` ?

**Comment by AndreiEres** [comment]
Look at that please as well https://github.com/paritytech/polkadot-sdk/actions/runs/10166554064/job/28141652578?pr=5065
There are suggestions for the pr_doc

**Review by acatangiu — APPROVED** [review]

**Review by AndreiEres — APPROVED** [review]
Good job @Safrout1, thank you

**Review by sandreim — APPROVED** [review]

**Comment by AndreiEres** [comment]
@Safrout1 could you provide your DOT address in the PR description?

**Comment by Safrout1** [comment]
> @Safrout1 could you provide your DOT address in the PR description?
 
@AndreiEres done, but would you like to explain why?

**Comment by AndreiEres** [comment]
> @AndreiEres done, but would you like to explain why?

I'll try to tip you for the your job

**Pull request merged**
